include/audio_module/audio_module.h

```diff
@@ -0,0 +1,500 @@
+/*
+ * Copyright(c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#ifndef _AUDIO_MODULE_H_
+#define _AUDIO_MODULE_H_
+
+/**
+ * @file
+ * @defgroup audio_module Audio module
+ * @{
+ * @brief Audio module for LE Audio applications.
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <zephyr/kernel.h>
+
+#include "data_fifo.h"
+#include "audio_defines.h"
+
+/**
+ * @brief Number of valid location bits.
+ *
+ */
+#define AUDIO_MODULE_LOCATIONS_NUM (32)
+
+/**
+ * @brief Module type.
+ *
+ */
+enum audio_module_type {
+	/* The module type is undefined. */
+	AUDIO_MODULE_TYPE_UNDEFINED = 0,
+
+	/* This is an input processing module.
+	 *
+	 * @note An input module obtains data internally within the
```

Comment:
I dont think it is needed to have @note here, same for the ones below

Reply:
This is important and has removed confusion.

Can we discuss?

Reply:
I vote that we keep these.

Reply:
I only meant the actual "@note", and not the text. But we can keep it like this